edge-modules/DirectMethodSender/Program.cs

```diff
@@ -29,12 +29,22 @@ public static async Task<int> MainAsync()
                     ModuleUtil.DefaultTransientRetryStrategy,
                     Logger);
 
-                Uri analyzerUrl = Settings.Current.AnalyzerUrl;
-                AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = analyzerUrl.AbsoluteUri };
-
                 (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
 
-                await CallDirectMethod(moduleClient, analyzerClient, Settings.Current.DirectMethodDelay, cts);
+                string analyzerUrl = Settings.Current.AnalyzerUrl;
+                if (analyzerUrl != string.Empty)
+                {
+                    Uri analyzerUri = new Uri(analyzerUrl);
+                    AnalyzerClient analyzerClient = new AnalyzerClient { BaseUrl = analyzerUri.AbsoluteUri };
+                    Action<MethodResponse> reportResult = async (response) => await ReportStatus(Settings.Current.TargetModuleId, response, analyzerClient);
+                    await StartDirectMethdTests(moduleClient, reportResult, Settings.Current.DirectMethodDelay, cts);
+                }
+                else
+                {
+                    Action<MethodResponse> reportResult = async (response) => await moduleClient.SendEventAsync("AnyOutput", new Message(Encoding.UTF8.GetBytes("Direct Method Call succeeded.")));
+                    await StartDirectMethdTests(moduleClient, reportResult, Settings.Current.DirectMethodDelay, cts);
+                }
+
                 await moduleClient.CloseAsync();
```

Comment:
move this line inside else block

Reply:
module client is needed in both cases

Reply:
It is passed into the StartDirectMethodTests()

Reply:
but both cases are inside the try, right?